fix  tests

rdinicut opened the pull request:
We should at some point update the fields to be camelCase instead of snake case. The only issue right now is the agreement_id but it is kindOf error prone. Why did we switch to camelCase btw? We were generating snake case from swagger hub using the code gen options?

Commit:
fix tests

charleenfei commented:
> We should at some point update the fields to be camelCase instead of snake case. The only issue right now is the agreement_id but it is kindOf error prone. Why did we switch to camelCase btw? We were generating snake case from swagger hub using the code gen options?

i couldn't get snake case generation functioning properly with swagger code-gen in a timeboxed way, there is an open issue on this #442. the node expects snake_cased attributes.

rdinicut commented:
@charleenfei can I merge this?

charleenfei commented:
I think we should keep it snake_cased and use ts-ignore instead, like how we treat all the other cases in the other controllers, so it is consistent. rn we don't use funding agreements in gw but if we did use them we would need to snake case, so I think we should at least keep it functional.

rdinicut commented:
so the main problem is here https://github.com/centrifuge/gateway/blob/c88be51afea54ab9d93464a0fdc7818b312942b7/packages/server/src/funding/funding.controller.ts#L65
.
 So the request will return snake case and this is a bug?

rdinicut commented:
@charleenfei @mikiquantum can you add me to swagger hub https://app.swaggerhub.com/apis/centrifuge.io/cent-node/2.0.0 so I can see the generation options?

Commit:
update api.ts snake_case, refactor and fix proxy error

rdinicut commented:
cc @charleenfei I updated the api with snake case and removed all ts-ignores and refactor back to snake case. Can you pull this branch delete all node modules using something like  ``` find . -name "node_modules" -type d -prune -exec rm -rf '{}' +```, yarn install and yarn start?

rdinicut commented on packages/ui/src/documents/CreateDocument.tsx:
```diff
@@ -83,13 +83,14 @@ export const CreateDocument: FunctionComponent<Props> = (props) => {
         } else {
           await httpClient.documents.create(document);
         }
+        go(0);
```
I think this was the problem with the proxy error. Also we were not getting the alerts from the document errors. We can improve this for sure but I think it is a different topic

rdinicut commented on packages/ui/src/documents/CreateDocument.tsx:
```diff
@@ -119,7 +120,7 @@ export const CreateDocument: FunctionComponent<Props> = (props) => {
         return <SecondaryHeader>
           <Box direction="row" gap="small" align="center">
-            <Link to={documentRoutes.index} size="large">
+            <Link to={documentRoutes.index}>
```
Bug, This was generating typescript errors for because Link that not have that prop

rdinicut commented on packages/server/src/centrifuge-client/centrifuge.module.ts:
```diff
@@ -8,10 +8,6 @@ function checkNodeEnvironment() {
     case 'test': {
       return new MockCentrifugeService();
     }
-    case 'functional': {
```
This was also generating typescript error. I assumed we are not using this so I removed it. If we want to add e2e test we can figure it out then

Commit:
change import
